server/routes/detail/detailController.test.ts

```diff
@@ -7,97 +7,80 @@ import auditServiceMock from '../../testutils/auditServiceMock'
   let req: Request
   let res: Response
-  let controller: DetailController
 
   const detail = { prisonNumber: 'AB12345', personalDetails: { firstName: 'JOHN', lastName: 'SMITH' } }
 
   beforeEach(() => {
     jest.resetAllMocks()
     req = {
-      query: { page: '0', sort: 'dateCreated,ASC', alertType: 'R', from: '01/01/2023', to: '02/02/2023' },
-      path: 'alerts/active',
-      user: { token: 'token', username: 'user', authSource: 'nomis' },
+      user: { token: 'token' },
       session: {},
       body: {},
     } as unknown as Request
     res = {
-      locals: {
-        user: {
-          authSource: 'nomis',
-          userRoles: [],
-          token: 'TOKEN',
-          username: 'user',
-          userId: '12345',
-          name: 'Billy Bob',
-          displayName: 'Billy Bob',
-        },
-      },
+      locals: { user: { username: 'user' } },
       render: jest.fn(),
       redirect: jest.fn(),
       status: jest.fn(),
     } as unknown as Response
-    controller = new DetailController(historicalPrisonerService, auditService)
   })
 
   describe('getDetail', () => {
     it('should retrieve prisoner detail and render the detail page if no session object', async () => {
       historicalPrisonerService.getPrisonerDetail.mockResolvedValue(detail)
+      req.params = { prisonNo: 'AB12345' }
 
-      await controller.getDetail(
-        { ...req, params: { prisonNo: 'AB12345' } } as unknown as Request,
-        { ...res } as Response,
-      )
+      await controller.getDetail(req, res)
 
       expect(historicalPrisonerService.getPrisonerDetail).toHaveBeenCalledWith('token', 'AB12345')
-      expectDetailContainingDetail()
+      expectRenderDetailContainingDetail()
+      expect(req.session.prisonerDetail).toBe(detail)
     })
+
     it('should retrieve prisoner detail and render the detail page if session object contains different prisoner', async () => {
       historicalPrisonerService.getPrisonerDetail.mockResolvedValue(detail)
+      req.params = { prisonNo: 'BC12345' }
+      req.session.prisonerDetail = detail
 
-      await controller.getDetail(
-        { ...req, params: { prisonNo: 'BC12345' }, session: { prisonerDetail: detail } } as unknown as Request,
-        { ...res } as Response,
-      )
+      await controller.getDetail(req, res)
 
       expect(historicalPrisonerService.getPrisonerDetail).toHaveBeenCalledWith('token', 'BC12345')
-      expectDetailContainingDetail()
+      expectRenderDetailContainingDetail()
+      expect(req.session.prisonerDetail).toBe(detail)
     })
 
     it('should just render the detail page if session object exists', async () => {
-      await controller.getDetail(
-        { ...req, params: { prisonNo: 'AB12345' }, session: { prisonerDetail: detail } } as unknown as Request,
-        { ...res } as Response,
-      )
+      req.params = { prisonNo: 'AB12345' }
+      req.session.prisonerDetail = detail
+
+      await controller.getDetail(req, res)
```

Comment:
switch to passing through `req` and `res` rather than destructuring, which is confusing and means we can't then assert things on the session in the test